pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/groupby/NoDictionarySingleColumnGroupKeyGenerator.java

```diff
@@ -72,47 +78,122 @@ public int getGlobalGroupKeyUpperBound() {
   @Override
   public void generateKeysForBlock(TransformBlock transformBlock, int[] groupKeys) {
     BlockValSet blockValSet = transformBlock.getBlockValueSet(_groupByExpression);
+    RoaringBitmap nullBitmap = blockValSet.getNullBitmap();
```

Comment:
This should be under the `nullHandlingEnabled` branch. I feel it is cleaner if we make 2 helper method, one for existing logic, one for null handling

Reply:
indeed. Split into 2 helper methods.